lib/instrumentation/transaction.js

```diff
@@ -13,77 +12,97 @@ var Timer = require('./timer')
 
 module.exports = Transaction
 
-function Transaction (agent, name, type) {
-  Object.defineProperty(this, 'name', {
-    configurable: true,
-    enumerable: true,
-    get () {
-      // Fall back to a somewhat useful name in case no _defaultName is set.
-      // This might happen if res.writeHead wasn't called.
-      return this._customName ||
-        this._defaultName ||
-        (this.req ? this.req.method + ' unknown route (unnamed)' : 'unnamed')
-    },
-    set (name) {
-      if (this.ended) {
-        agent.logger.debug('tried to set transaction.name on already ended transaction %o', { id: this.id })
-        return
-      }
-      agent.logger.debug('setting transaction name %o', { id: this.id, name: name })
-      this._customName = name
-    }
-  })
-
-  Object.defineProperty(this, 'result', {
-    configurable: true,
-    enumerable: true,
-    get () {
-      return this._result
-    },
-    set (result) {
-      if (this.ended) {
-        agent.logger.debug('tried to set transaction.result on already ended transaction %o', { id: this.id })
-        return
-      }
-      agent.logger.debug('setting transaction result %o', { id: this.id, result: result })
-      this._result = result
-    }
-  })
-
-  Object.defineProperty(this, 'timestamp', {
-    get () {
-      if (!this._timestamp) {
-        this._timestamp = new Date(this._timer.start).toISOString()
-      }
-      return this._timestamp
-    }
-  })
+function Transaction (agent, name, type, traceparent) {
+  this.context = TraceContext.startOrResume(traceparent, agent._conf)
+  const verb = this.context.parentId ? 'continue' : 'start'
+  agent.logger.debug('%s transaction %o', verb, { id: this.id, name: name, type: type })
```

Comment:
Would it be more correct to say "start/continue trace" instead of "start/continue transaction"?

Reply:
fixed.